common-ui-compose/src/main/java/app/tivi/common/compose/Insets.kt

```diff
@@ -0,0 +1,300 @@
+ * )
+ * ```
+ *
+ * Internally this uses [Modifier.height] so has the same characteristics with regards to incoming
+ * layout constraints.
+ */
+fun Modifier.statusBarHeight() = composed {
+    // TODO: Move to Android 11 WindowInsets APIs when they land in AndroidX.
+    // It currently assumes that status bar == top which is probably fine, but doesn't work
+    // in multi-window, etc.
+    val insets = InsetsAmbient.current
+    Modifier.height(insets.systemBars.top)
+}
+
+/**
+ * Declare the preferred height of the content to match the height of the navigation bar when present at the bottom of the screen.
+ *
+ * This is very handy when used with `Spacer` to push content below the navigation bar:
+ * ```
+ * Column {
+ *     // Content to be drawn above status bar (y-axis)
+ *     Spacer(Modifier.navigationBarHeight())
+ * }
+ * ```
+ *
+ * It's also useful when used with `Box` to draw a scrim which matches the navigation bar:
+ * ```
+ * Box(
+ *     Modifier.navigationBarHeight()
+ *         .fillMaxWidth()
+ *         .drawBackground(MaterialTheme.colors.background.copy(alpha = 0.3f)
+ * )
+ * ```
+ *
+ * Internally this uses [Modifier.height] so has the same characteristics with regards to incoming
+ * layout constraints.
+ */
+fun Modifier.navigationBarHeight() = composed {
+    // TODO: Move to Android 11 WindowInsets APIs when they land in AndroidX.
+    // It currently assumes that nav bar == bottom, which is wrong in landscape.
+    // It also doesn't handle the IME correctly.
+    val insets = InsetsAmbient.current
+    Modifier.height(insets.systemBars.bottom)
+}
+
+enum class HorizontalSide { Left, Right }
+
+/**
+ * Declare the preferred width of the content to match the width of the navigation bar,
+ * on the given [side].
+ *
+ * This is very handy when used with `Spacer` to push content inside from any vertical
+ * navigation bars (typically when the device is in landscape):
+ * ```
+ * Row {
+ *     Spacer(Modifier.navigationBarWidth(HorizontalSide.Left))
+ *
+ *     // Content to be inside the navigation bars (x-axis)
+ *
+ *     Spacer(Modifier.navigationBarWidth(HorizontalSide.Right))
+ * }
+ * ```
+ *
+ * It's also useful when used with `Box` to draw a scrim which matches the navigation bar:
+ * ```
+ * Box(
+ *     Modifier.navigationBarWidth(HorizontalSide.Left)
+ *         .fillMaxHeight()
+ *         .drawBackground(MaterialTheme.colors.background.copy(alpha = 0.3f)
+ * )
+ * ```
+ *
+ * Internally this uses [Modifier.width] so has the same characteristics with regards to incoming
+ * layout constraints.
+ *
+ * @param side The navigation bar side to use as the source for the width.
+ */
+fun Modifier.navigationBarWidth(side: HorizontalSide) = composed {
+    // TODO: Move to Android 11 WindowInsets APIs when they land in AndroidX.
+    // It currently assumes that nav bar == left/right
+    val insets = InsetsAmbient.current
+    when (side) {
+        HorizontalSide.Left -> Modifier.width(insets.systemBars.left)
```

Comment:
See other comments about staying in pixels and deferring the state reads to layout

Reply:
Done.